modules/openthread/platform/messagepool.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <openthread/platform/messagepool.h>
+
+#define LOG_MODULE_NAME net_otPlat_messagepool
+#define LOG_LEVEL       CONFIG_OPENTHREAD_LOG_LEVEL
+
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define BUF_TIMEOUT K_MSEC(50)
+
+#define MESSAGE_POOL_SIZE                                                                          \
+	(CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS * CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE)
+#define MAX_ALIGNMENT _Alignof(z_max_align_t)
+
+BUILD_ASSERT(CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE % MAX_ALIGNMENT == 0,
+	     "Invalid message buffer size");
+
+static struct k_mem_slab message_pool;
+_Alignas(MAX_ALIGNMENT) static uint8_t message_pool_buffer[MESSAGE_POOL_SIZE];
```

Comment:
`__aligned()`, `__alignof__()` are commonly used in Zephyr codebase, we should perhaps stick to that for consistency.